xcodeutil/testsummaries.go

```diff
@@ -6,9 +6,92 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/bitrise-io/go-utils/fileutil"
 	"github.com/bitrise-tools/go-xcode/plistutil"
 )
 
+// const ...
+const (
+	OldTestSummaries TestSummaryType = "OldTestSummaries"
+	NewTestSummaries TestSummaryType = "NewTestSummaries"
+)
+
+// TestSummaryType ...
+type TestSummaryType string
+
+// TestSummaries ...
+type TestSummaries struct {
+	Version                  TestSummaryType
+	Content                  string
+	TestItemsWithScreenshots []map[string]interface{}
+}
+
+// New returns a TestSummaries from the provided *_TestSummaries.plist's path.
+// It will search for test items with screenhot and it will set the version of the TestSummaries (OldTestSummaries/NewTestSummaries) depending on that the plist has `HasScreenshotData` fields or `Attachments` fileds.
+func New(testSummariesPth string) (*TestSummaries, error) {
+	var testSummaries TestSummaries
+	var err error
+
+	testSummaries.Content, err = fileutil.ReadStringFromFile(testSummariesPth)
+	if err != nil {
+		return nil, err
+	}
+
+	testSummaries, err = testSummaries.collectTestItemsWithScreenshotAndSetVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	return &testSummaries, nil
+}
+
+func (t TestSummaries) collectTestItemsWithScreenshotAndSetVersion() (TestSummaries, error) {
+	testSummaryType := OldTestSummaries
+
+	testSummariesPlistData, err := plistutil.NewPlistDataFromContent(t.Content)
+	if err != nil {
+		return t, err
+	}
+
+	testableSummaries, err := getValueAsMapStringInterfaceArray(testSummariesPlistData, "TestableSummaries")
+	if err != nil {
+		return t, err
+	}
+
+	subActivitiesWithScreenshot := []map[string]interface{}{}
+	for _, testableSummariesItem := range testableSummaries {
+		tests, err := getValueAsMapStringInterfaceArray(testableSummariesItem, "Tests")
+		if err != nil {
+			return t, err
+		}
+
+		for _, testsItem := range tests {
+			lastSubtests, err := collectLastSubtests(testsItem)
+			if err != nil {
+				return t, err
+			}
+
+			for _, lastSubtest := range lastSubtests {
```

Comment:
are you sure that this is the most depth of the subSummaries, should not it use recursion just like the old one?